services/datamanager/builtin/builtin.go

```diff
@@ -408,22 +419,15 @@ func (svc *builtIn) initSyncer(ctx context.Context) error {
 	ctx, cancel := context.WithTimeout(ctx, grpcConnectionTimeout)
 	defer cancel()
 	identity, conn, err := svc.cloudConnSvc.AcquireConnection(ctx)
-	if errors.Is(err, cloud.ErrNotCloudManaged) {
```

Comment:
Previously if a user used a local config data manager just wouldn't initialize if they configured data sync. This changes it so that datamanger will initialize with local configs, data sync just won't do anything.

Reply:
Would it throw an error here when it was a local config? Or below when calling svc.syncerConstructor? I would have expected that this would return a Noop syncer and thus still continue with a valid data manager service that captures.

Reply:
Previously if you booted viam-server with a local config that used data sync it would return an error here: https://github.com/viamrobotics/rdk/pull/4164/files#diff-ffb86a3e2b364cc470f33937bb1ffb08ca2ef9d01b2734320daea7bee24a8943L416 which would result in the Reconfigure method returning an error which would result in data management never booting.

Reply:
The fact that we set `svc.syncer = datasync.NewNoopManager()` doesn't change the fact that `initSyncer` returns an error.

Reply:
What error would be returned if it was a local config? Wondering if we could instead add if errors.Is(err, cloud.ErrNotCloudManaged) or _whatever the error is when it's local_.

I ask this since we modeled the noop sync manager after the [noop package manager](https://github.com/viamrobotics/rdk/blob/a0b1b46751ae39cfa5b6ee60d460299360cb2f34/robot/packages/noop_package_manager.go#L10) that we use in RDK to download packages.

Reply:
`ErrNotCloudManaged` is the error returned if it is a local config.
I think we are doing `if errors.Is(err, cloud.ErrNotCloudManaged) {` here: https://github.com/viamrobotics/rdk/pull/4164/files#diff-ffb86a3e2b364cc470f33937bb1ffb08ca2ef9d01b2734320daea7bee24a8943R681 which will terminate the goroutine trying to boot data capture: https://github.com/viamrobotics/rdk/pull/4164/files#diff-ffb86a3e2b364cc470f33937bb1ffb08ca2ef9d01b2734320daea7bee24a8943R659 as the viam-server using a local config means data sync will NEVER succeed for that viam server instance and as a result we should give up on trying to boot data captuer.

Does that seem reasonable?

Reply:
I don't understand what the noop package manager has to do with whether or not we give up on trying to boot data sync depending on if it is possible for viam-server to connect to the cloud.